packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -1,20 +1,43 @@
-import { css } from '@emotion/css';
+import { css, cx } from '@emotion/css';
 import React, { CSSProperties, ReactNode } from 'react';
 
-import { GrafanaTheme2 } from '@grafana/data';
+import { GrafanaTheme2, LoadingState } from '@grafana/data';
 
 import { useStyles2, useTheme2 } from '../../themes';
+import { IconName } from '../../types/icon';
+import { IconButton } from '../IconButton/IconButton';
+import { PopoverContent } from '../Tooltip';
+
+/**
+ * @internal
+ */
+interface PanelChromeInfoState {
+  icon: IconName;
+  tooltip?: PopoverContent;
+  onClick?: () => void;
+}
 
 /**
  * @internal
  */
 export interface PanelChromeProps {
   width: number;
   height: number;
-  title?: string;
+  children: (innerWidth: number, innerHeight: number) => ReactNode;
   padding?: PanelPadding;
-  leftItems?: React.ReactNode[]; // rightItems will be added later (actions links etc.)
-  children: (innerWidth: number, innerHeight: number) => React.ReactNode;
+  title?: string;
+  titleItems?: PanelChromeInfoState[];
+  menu?: ReactNode;
+  dragClass?: string;
+  hoverHeader?: boolean;
+  loadingState?: LoadingState;
+  states?: ReactNode[];
```

Comment:
What's the difference between these two?

Reply:
As far as I understand, the `states` prop carries any the following (ordered by priority): Error, Warning, Data Info, Streaming.
Whereas the `loadingState` shows as 
![image](https://user-images.githubusercontent.com/13227501/202177487-01d50972-f777-420d-bb9f-5c3f1a1d0ae9.png)
or 
![image](https://user-images.githubusercontent.com/13227501/202177511-26e3fac1-3bfe-4c6d-8da7-67c1312cceba.png)
while something is loading, not interrupting the "Data Info" or "Streaming" states that might be passed as nodes. 

Honestly, that's my interpretation. I believe @torkelo can give better feedback if I've understood those correctly.